### PR TITLE
feat(application-system): Be able to configure if applications are listed and/or pruned by state

### PR DESCRIPTION
**Pull request by baering**
# \<Description\>

This PR adds a new _required_ `meta.lifecycle` on application states.

If you are not a part of `aranja` or `core` then feel free to skip reviewing code that is not owned by your team, but of course all reviews are appreciated. It should just be a few lines adding `meta.lifecycle` to your application template with the default value set.

## What

At the moment it contains two main properties:
- `shouldBeListed`: should the application be visible in my pages and/or `/umsoknir/:type` overview screen
- `shouldBePruned` should the application be pruned when it has stayed in this state for a certain amount of time, if so `whenToPrune` also needs to be defined, either time in milliseconds or a callback that returns a date

This PR does not address the actual pruning of applications and their attachments, that part will be handled separately. But they will get filtered out, so that when Date.now > prune time those applications cannot be fetched.

## Why

To allow application developers to create a `prerequisite` state where applications are not formally "created" but a place to start an application.

This could be a state to:
- validate some conditions/requirements before starting an application
- route the user to existing applications

When the user advances from this prerequisite state the application has "started" and will be listed in the appropriate places.

## Screenshots / Gifs

Attach Screenshots / Gifs to help reviewers understand the scope of the pull request

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

File: apps/application-system/api/src/app/modules/application/application.controller.ts [hunk 1]
```diff
@@ -164,6 +165,7 @@ export class ApplicationController {
       // We've already checked an application with this type and it is ready
       if (templateTypeToIsReady[application.typeId]) {
         filteredApplications.push(application)
+        continue
```
Comment (baering) [inline apps/application-system/api/src/app/modules/application/application.controller.ts:168]: This is unrelated but fixes an issue with duplicate listings of applications

File: apps/application-system/api/src/app/modules/application/application.controller.ts [hunk 2]
```diff
@@ -237,7 +239,24 @@ export class ApplicationController {
...
+    const createdApplication = await this.applicationService.create(
+      applicationDto,
+    )
+
+    // Make sure the application has the correct lifecycle values persisted to database.
+    // Requires an application object that is created in the previous step.
+    const {
+      updatedApplication,
+    } = await this.applicationService.updateApplicationState(
+      createdApplication.id,
```
Comment (jeremybarbet) [inline apps/application-system/api/src/app/modules/application/application.controller.ts:251]: Not related to your code, but I wonder if passing the Application object would be easier than each needed property as parameter

File: apps/application-system/api/src/app/modules/application/application.controller.ts [hunk 3]
```diff
@@ -378,6 +397,7 @@ export class ApplicationController {
     @AuthorizationHeader() authorization: string,
     @NationalId() nationalId: string,
   ): Promise<ApplicationResponseDto> {
+    await new Promise((resolve) => setTimeout(resolve, 1000))
```
Comment (jeremybarbet) [inline apps/application-system/api/src/app/modules/application/application.controller.ts:400]: Still needed?

File: libs/application/templates/parental-leave/src/forms/Prerequisites.ts [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+  buildSubSection,
+  Form,
+  FormModes,
+} from '@island.is/application/core'
+
+import { parentalLeaveFormMessages } from '../lib/messages'
+import Logo from '../assets/Logo'
+
+export const PrerequisitesForm: Form = buildForm({
+  id: 'ParentalLeaveDraft',
```
Comment (jeremybarbet) [inline libs/application/templates/parental-leave/src/forms/Prerequisites.ts:20]: Should probably have a different id than the main template

File: libs/application/templates/parental-leave/src/forms/Prerequisites.ts [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+            }),
+          ],
+        }),
+        buildSubSection({
+          id: 'chooseBaby',
+          title: 'Veldu barn',
+          children: [
+            buildMultiField({
+              id: 'selectedBaby',
+              title: 'Veldu barn',
```
Comment (jeremybarbet) [inline libs/application/templates/parental-leave/src/forms/Prerequisites.ts:93]: Missing some translations on a few titles/descriptions

**Review by jeremybarbet — COMMENTED** [review]
Only a few minor comments, looks good to me otherwise

**Review by Rikkiabb — APPROVED** [review]
LGTM 🚀

File: libs/application/templates/parental-leave/src/forms/Prerequisites.ts (hunk 4 above)
Comment (baering) [reply libs/application/templates/parental-leave/src/forms/Prerequisites.ts:20]: Copy paste, my bad 💩  thanks!

File: apps/application-system/api/src/app/modules/application/application.controller.ts (hunk 3 above)
Comment (baering) [reply apps/application-system/api/src/app/modules/application/application.controller.ts:400]: Definitely not, should have removed. Thanks.

**Review by fjandakornid — APPROVED** [review]
LGTM

**Review by stjanilofts — APPROVED** [review]

**Review by dd-sdang — APPROVED** [review]
Nice feature! Health Insurance looks good :)

**Review by atligudl — APPROVED** [review]

**Review by jeremybarbet — APPROVED** [review]
LGTM 🎉

**Pull request merged**